File: app/src/main/java/ua/hospes/rtm/data/team/mapper/TeamsMapper.java
package ua.hospes.rtm.data.team.mapper;

import android.database.Cursor;

import java.util.List;

import ua.hospes.rtm.core.db.tables.Teams;
import ua.hospes.rtm.data.team.models.TeamDb;
import ua.hospes.rtm.domain.drivers.models.Driver;
import ua.hospes.rtm.domain.team.models.Team;

/**
 * @author Andrew Khloponin
 */
public class TeamsMapper {
    public static TeamDb map(Cursor cursor) {
        int id = cursor.getInt(cursor.getColumnIndex(Teams._ID));
        String name = cursor.getString(cursor.getColumnIndex(Teams.NAME));
        return new TeamDb(id, name);
    }

    public static Team map(TeamDb db, List<Driver> drivers) {
        Team team = new Team(db.getId());
        team.setName(db.getName());
        team.setDrivers(drivers);
        return team;
    }

    public static TeamDb map(Team team) {
        return new TeamDb(team.getId(), team.getName());
    }
}

File: app/src/main/java/ua/hospes/rtm/data/team/TeamsMapper.kt
package ua.hospes.rtm.data.team

import android.database.Cursor
import ua.hospes.rtm.core.db.tables.Teams
import ua.hospes.rtm.domain.drivers.Driver
import ua.hospes.rtm.domain.team.Team

object TeamsMapper {
    @JvmStatic fun map(cursor: Cursor): TeamDb = TeamDb(
            cursor.getInt(cursor.getColumnIndex(Teams.ID.name())),
            cursor.getString(cursor.getColumnIndex(Teams.NAME.name()))
    )

    @JvmStatic fun map(db: TeamDb, drivers: List<Driver>): Team = Team(db.id, db.name, drivers.toMutableList())

    @JvmStatic fun map(team: Team): TeamDb = TeamDb(team.id, team.name)
}